### ISSUE TITLE
Ctrl+Shift+F does not work correctly

### ISSUE THREAD

**Issue by friederbluemle** [rlogiacco/Natural#24]
When applying the auto-formatter, this plugin does not correctly format Gherkin feature files.

Before:
![image](https://f.cloud.github.com/assets/743291/1040292/ccb0805c-0fb1-11e3-90a4-1d7b9ee7f00d.png)

After:
![image](https://f.cloud.github.com/assets/743291/1040305/02fa9652-0fb2-11e3-9664-3401389c4db2.png)

Note the incorrect indentation, mix of tabs and spaces, and appended spaces.

Is there a way to customize the formatter or does this require a fix in the plugin iteself?
##

**Comment by friederbluemle** [comment]
Thanks @github-actions 😂

**Comment by drkstr101** [comment]
This is fixable using the new `formatting2` API in latest Xtext.

I am flagging this issue for review in [next](https://github.com/rlogiacco/Natural/pull/73)

**Comment by friederbluemle** [comment]
Can we please keep this open until the pending work/feedback is resolved?

**Comment by drkstr101** [comment]
@rlogiacco I agree, this is still an unresolved issue. It has improved somewhat in current master (unreleased as of yet), but won't be fully supported until the 1.0 milestone release.

The current behavior in master is to simply preserve the existing Narrative formatting. Later we can add rules for things like whitespace cleanup or auto-indent.

**Comment by drkstr101** [comment]
@rlogiacco This can be closed with https://github.com/rlogiacco/Natural/pull/86
